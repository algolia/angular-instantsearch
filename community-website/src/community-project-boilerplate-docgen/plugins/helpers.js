const highlight = require('./syntaxHighlighting.js');
const md = require('./mdRenderer.js');

// this plugin provides ATM one helper to easily compute the publicPath of assets
module.exports = function helpers(filenames, metalsmith, cb) {
  // eslint-disable-next-line no-param-reassign
  metalsmith.metadata().h = {
    markdown(src) {
      return md.render(src);
    },
    highlight(src, opts) {
      const lang = opts && opts.lang;
      return highlight(src, lang);
    },
    maybeActive(navPath, singlePathOrArrayOfPaths) {
      const pathsToTest = [].concat(singlePathOrArrayOfPaths);
      return pathsToTest.some(pathToTest => navPath.indexOf(pathToTest) === 0)
        ? 'active'
        : '';
    },
  };

  cb();
};
