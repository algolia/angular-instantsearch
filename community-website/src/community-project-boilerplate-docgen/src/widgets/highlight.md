---
title: Widgets - Highlight
layout: widget.pug
---

# Highlight

## Description

Renders any attribute from an hit into its highlighted form when relevant.

## Options

| Attributes  | Type     | Description
| -           | -        | -
| `attribute` | `string` | The location of the highlighted attribute in the hit
| `hit`       | `{}`     | The hit object containing the highlighted attribute
| `tagName?`  | `string` | The tag to be used for highlighted parts of the hit

## Code example

```html
<ng-ais-hits>
  <ng-template let-hits="hits">
    <div *ngFor="let hit of hits">
      Hit {{hit.objectID}}:
      <ng-ais-highlight attribute="name" [hit]="hit">
      </ng-ais-highlight>
    </div>
  </ng-template>
</ng-ais-hits>
```
