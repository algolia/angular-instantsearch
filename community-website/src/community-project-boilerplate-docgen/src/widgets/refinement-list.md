---
title: Widgets - Refinement List
layout: widget.pug
canonical: https://www.algolia.com/doc/api-reference/widgets/refinement-list/angular/
---

# Refinement List

## Description

The RefinementList component displays a list that let the end user choose multiple values for a specific facet.

## Options

| Attribute            | Type                     | Description
| -                    | -                        | -
| `attribute`          | `string`                 | Name of the attribute for faceting (eg. "categories")
| `operator?`          | `"and" / "or"`           | How to apply the refinements
| `limit?`             | `number`                 | How much facet values to get
| `showMoreLimit?`     | `number`                 | Bigger than `limit` if the component should display a button that will expand the number of items
| `showMoreLabel?`     | `string`                 | Label of the show more button
| `showLessLabel?`     | `string`                 | Label of the show less button
| `searchable?`        | `boolean`                | True if the component should display an input to search for facet values
| `searchPlaceholder?` | `string`                 | The label of the search input placeholder
| `sortBy?`            | `string[] / Function`    | How to sort facet values
| `autoHideContainer?` | `boolean`                | Hides the refinement list if there's no item to display
| `transformItems?`    | `(items) => items`       | Function to modify the items being displayed, e.g. for filtering or sorting them

## Code example

```html
<ais-refinement-list
  attribute="brand"
  operator="or"
  [limit]="10"
  [transformItems]="translate"
>
</ais-refinement-list>
```
```ts
export class AppComponent {
  // ...
  translate(items) {
    return items.map(item => ({
      ...item,
      highlighted: myTranslateFunc(item.highlighted)
    }));
  }
}
```
