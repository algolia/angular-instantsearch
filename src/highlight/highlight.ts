import { Component, Input } from "@angular/core";
import { isPlainObject, escape, get } from "lodash-es";
import { bem } from "../utils";

@Component({
  selector: "ais-highlight",
  template: `<span [class]="cx()" [innerHtml]="content"></span>`
})
export class NgAisHighlight {
  @Input() attribute: string;
  @Input() hit: { _highlightResult?: {}; label?: string; highlighted?: string };
  @Input() tagName: string = "em";

  cx = bem("Highlight");

  get content() {
    if (this.attribute === "highlighted") {
      return this.hit.highlighted
        ? this.replaceWithTagName(this.hit.highlighted)
        : this.hit.label;
    }

    if (this.hit.hasOwnProperty("_highlightResult")) {
      const attributeHighlighted = get(
        this.hit._highlightResult,
        this.attribute
      );

      // check that the attributeHighlighted is a string
      if (
        isPlainObject(attributeHighlighted) &&
        typeof attributeHighlighted.value === "string"
      ) {
        return this.replaceWithTagName(attributeHighlighted.value);
      }
    }

    const fallback = get(this.hit, this.attribute);
    if (!fallback) {
      console.warn(
        `Could not find attribute [${
          this.attribute
        }] into hit object, will display an empty string.`
      );

      return "";
    }

    return fallback;
  }

  replaceWithTagName(value: string) {
    return value
      .replace(
        new RegExp("<em>", "g"),
        `<${this.tagName} class="${this.cx("highlighted")}">`
      )
      .replace(new RegExp("</em>", "g"), `</${this.tagName}>`);
  }
}
