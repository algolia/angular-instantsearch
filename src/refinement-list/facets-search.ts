import { Component, Input } from '@angular/core';
import { bem } from '../utils';

@Component({
  selector: 'ais-facets-search',
  template: `
    <div [class]="cx()">
      <form
        [class]="cx('form')"
        (submit)="handleSubmit($event)"
        novalidate
      >
        <input
          [class]="cx('input')"
          autocapitalize="off"
          autocorrect="off"
          placeholder="{{searchPlaceholder}}"
          role="textbox"
          spellcheck="false"
          type="text"
          [value]="searchQuery"
          (input)="handleChange($event.target.value)"
        />

        <button
          [class]="cx('submit')"
          title="Submit the search query."
          type="submit"
        >
          <svg
            [ngClass]="cx('submitIcon')"
            viewBox="0 0 40 40"
            width="10"
            height="10"
          >
            <path d="M26.804 29.01c-2.832 2.34-6.465 3.746-10.426 3.746C7.333 32.756 0 25.424 0 16.378 0 7.333 7.333 0 16.378 0c9.046 0 16.378 7.333 16.378 16.378 0 3.96-1.406 7.594-3.746 10.426l10.534 10.534c.607.607.61 1.59-.004 2.202-.61.61-1.597.61-2.202.004L26.804 29.01zm-10.426.627c7.323 0 13.26-5.936 13.26-13.26 0-7.32-5.937-13.257-13.26-13.257C9.056 3.12 3.12 9.056 3.12 16.378c0 7.323 5.936 13.26 13.258 13.26z"></path>
          </svg>
        </button>

        <button
          [class]="cx('reset')"
          type="reset"
          title="Clear the search query."
          hidden
        >
          <svg
            [ngClass]="cx('resetIcon')"
            viewBox="0 0 20 20"
            width="10"
            height="10"
          >
            <path d="M8.114 10L.944 2.83 0 1.885 1.886 0l.943.943L10 8.113l7.17-7.17.944-.943L20 1.886l-.943.943-7.17 7.17 7.17 7.17.943.944L18.114 20l-.943-.943-7.17-7.17-7.17 7.17-.944.943L0 18.114l.943-.943L8.113 10z"></path>
          </svg>
        </button>
      </form>
    </div>
  `,
})
export class NgAisFacetsSearch {
  @Input() public searchPlaceholder: string;
  @Input() public search: Function;

  public cx = bem('SearchBox');

  public searchQuery = '';

  public handleChange(value: string) {
    this.searchQuery = value;
    this.search(value);
  }

  public handleSubmit(event) {
    event.preventDefault();
    this.search(this.searchQuery);
  }
}
