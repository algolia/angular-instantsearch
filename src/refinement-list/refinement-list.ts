import { Component, Input, Inject, forwardRef } from '@angular/core';
import { connectRefinementList } from 'instantsearch.js/es/connectors';
import { BaseWidget } from '../base-widget';
import { NgAisInstantSearch } from '../instantsearch/instantsearch';
import { noop, parseNumberInput } from '../utils';

export type RefinementListItem = {
  value: string;
  label: string;
  count: number;
  isRefined: boolean;
};

export type RefinementListState = {
  canRefine: boolean;
  canToggleShowMore: boolean;
  createURL: Function;
  isShowingMore: boolean;
  items: RefinementListItem[];
  refine: Function;
  toggleShowMore: Function;
  searchForItems: Function;
  isFormSearch: boolean;
};

@Component({
  selector: 'ais-refinement-list',
  template: `
    <div
      [class]="cx()"
      *ngIf="!isHidden"
    >
      <div
        *ngIf="searchable"
        [class]="cx('searchBox')"
      >
        <ais-facets-search
          [search]="state.searchForItems"
          [searchPlaceholder]="searchPlaceholder"
        >
        </ais-facets-search>
      </div>

      <ul [class]="cx('list')">
        <li
          [class]="getItemClass(item)"
          *ngFor="let item of state.items"
          (click)="refine($event, item)"
        >
          <label [class]="cx('label')">
            <input
              [class]="cx('checkbox')"
              type="checkbox"
              value="{{item.value}}"
              [checked]="item.isRefined"
            />
            <span [class]="cx('labelText')">
              <ais-highlight attribute="highlighted" [hit]="item"></ais-highlight>
            </span>
            <span [class]="cx('count')">{{item.count}}</span>
          </label>
        </li>
      </ul>

      <button
        [class]="cx('showMore')"
        *ngIf="showMore"
        (click)="state.toggleShowMore()"
        [disabled]="!state.canToggleShowMore"
      >
        {{state.isShowingMore ? showLessLabel : showMoreLabel}}
      </button>
    </div>
  `,
})
export class NgAisRefinementList extends BaseWidget {
  // rendering options
  @Input() public showMoreLabel: string = 'Show more';
  @Input() public showLessLabel: string = 'Show less';
  @Input() public searchable?: boolean;
  @Input() public searchPlaceholder: string = 'Search here...';

  // instance options
  @Input() public attribute: string;
  @Input() public operator: 'or' | 'and';
  @Input() public limit: number;
  @Input() public showMore: boolean;
  @Input() public showMoreLimit: number;
  @Input() public sortBy: string[] | ((item: object) => number);
  @Input()
  public transformItems?: <U extends RefinementListItem>(
    items: RefinementListItem[]
  ) => U[];

  public state: RefinementListState = {
    canRefine: false,
    canToggleShowMore: false,
    createURL: noop,
    isShowingMore: false,
    items: [],
    refine: noop,
    toggleShowMore: noop,
    searchForItems: noop,
    isFormSearch: false,
  };

  get isHidden() {
    return this.state.items.length === 0 && this.autoHideContainer;
  }

  constructor(
    @Inject(forwardRef(() => NgAisInstantSearch))
    public instantSearchParent: any
  ) {
    super('RefinementList');
  }

  public ngOnInit() {
    this.createWidget(connectRefinementList, {
      showMore: this.showMore,
      limit: parseNumberInput(this.limit),
      showMoreLimit: parseNumberInput(this.showMoreLimit),
      attribute: this.attribute,
      operator: this.operator,
      sortBy: this.sortBy,
      escapeFacetValues: true,
      transformItems: this.transformItems,
    });

    super.ngOnInit();
  }

  public refine(
    event: MouseEvent,
    item: { isRefined: boolean; value: string }
  ) {
    event.preventDefault();
    event.stopPropagation();

    if (this.state.canRefine) {
      // update UI directly, it will update the checkbox state
      item.isRefined = !item.isRefined;

      // refine through Algolia API
      this.state.refine(item.value);
    }
  }
}
