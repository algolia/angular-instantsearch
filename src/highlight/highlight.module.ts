import { CommonModule } from '@angular/common';
import { NgModule } from '@angular/core';

import { NgAisHighlight } from './highlight';

@NgModule({
  declarations: [NgAisHighlight],
  entryComponents: [NgAisHighlight],
  exports: [NgAisHighlight],
  imports: [CommonModule],
})
export class NgAisHighlightModule {}
