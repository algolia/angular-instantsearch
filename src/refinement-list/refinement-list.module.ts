import { CommonModule } from "@angular/common";
import { NgModule } from "@angular/core";

import { NgAisHighlightModule } from "../highlight/highlight.module";
import { NgAisRefinementList } from "./refinement-list";

@NgModule({
  declarations: [NgAisRefinementList],
  entryComponents: [NgAisRefinementList],
  exports: [NgAisRefinementList],
  imports: [CommonModule, NgAisHighlightModule]
})
export class NgAisRefinementListModule {}
