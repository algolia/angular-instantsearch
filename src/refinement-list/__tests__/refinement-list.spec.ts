import { connectRefinementList } from 'instantsearch.js/es/connectors';
import { createRenderer } from '../../../helpers/test-renderer';
import { NgAisRefinementList } from '../refinement-list';
import { NgAisFacetsSearch } from '../facets-search';
import { NgAisHighlight } from '../../highlight/highlight';

const defaultState = {
  canRefine: true,
  canToggleShowMore: false,
  createURL: jest.fn(),
  isShowingMore: false,
  items: [
    { label: 'foo', count: 100, value: 'foo', isRefined: false },
    { label: 'bar', count: 100, value: 'bar', isRefined: false },
    { label: 'foobar', count: 100, value: 'foobar', isRefined: false },
    { label: 'barfoo', count: 100, value: 'barfoo', isRefined: false },
  ],
  refine: jest.fn(),
  toggleShowMore: jest.fn(),
};

const render = createRenderer({
  defaultState,
  template: '<ais-refinement-list></ais-refinement-list>',
  TestedWidget: NgAisRefinementList,
  additionalDeclarations: [NgAisHighlight, NgAisFacetsSearch],
});

describe('RefinementList', () => {
  it('renders markup without state', () => {
    const fixture = render();
    expect(fixture).toMatchSnapshot();
  });

  it('renders markup with state', () => {
    const fixture = render({});
    expect(fixture).toMatchSnapshot();
  });

  describe('connector init', () => {
    let render;
    let createWidget;
    beforeEach(() => {
      createWidget = jest.spyOn(NgAisRefinementList.prototype, 'createWidget');
      render = template =>
        createRenderer({
          defaultState,
          template,
          TestedWidget: NgAisRefinementList,
          additionalDeclarations: [NgAisHighlight, NgAisFacetsSearch],
        })({});
    });
    it('should be called with connectRefinementList', () => {
      render(`<ais-refinement-list></ais-refinement-list>`);
      expect(createWidget.mock.calls[0][0]).toEqual(connectRefinementList);
    });
    it('should be called with attributeName undefined by default', () => {
      render(`<ais-refinement-list></ais-refinement-list>\``);
      expect(createWidget.mock.calls[0][1].attribute).toBeUndefined();
    });
    it('should be called with attributeName passed down by attribute prop', () => {
      render(`<ais-refinement-list attribute="brands"></ais-refinement-list>`);
      expect(createWidget.mock.calls[0][1].attribute).toEqual('brands');
    });
    it('should be called with limit undefined by default', () => {
      render(`<ais-refinement-list></ais-refinement-list>`);
      expect(createWidget.mock.calls[0][1].limit).toEqual(undefined);
    });
    it('should be called with limit passed down as prop', () => {
      render(`<ais-refinement-list [limit]="30"></ais-refinement-list>`);
      expect(createWidget.mock.calls[0][1].limit).toEqual(30);
    });
    it('should be called with showMoreLimit undefined by default', () => {
      render(`<ais-refinement-list></ais-refinement-list>`);
      expect(createWidget.mock.calls[0][1].showMoreLimit).toBeUndefined();
    });
    it('should be called with showMoreLimit passed down as prop', () => {
      render(
        `<ais-refinement-list [showMoreLimit]="30"></ais-refinement-list>`
      );
      expect(createWidget.mock.calls[0][1].showMoreLimit).toEqual(30);
    });
    it('should be called with operator passed down as prop', () => {
      render(`<ais-refinement-list operator="and"></ais-refinement-list>`);
      expect(createWidget.mock.calls[0][1].operator).toEqual('and');
    });
    it('should be called with sortBy undefined by default', () => {
      render(`<ais-refinement-list></ais-refinement-list>`);
      expect(createWidget.mock.calls[0][1].sortBy).toBeUndefined();
    });
    it('should be called with sortBy passed down as prop', () => {
      render(`<ais-refinement-list sortBy="name"></ais-refinement-list>`);
      expect(createWidget.mock.calls[0][1].sortBy).toEqual('name');
    });
    it('should be called with sortBy passed down as prop', () => {
      render(
        `<ais-refinement-list [transformItems]="'func'"></ais-refinement-list>`
      );
      expect(createWidget.mock.calls[0][1].transformItems).toEqual('func');
    });
    it('should be called with escapeFacetValues true by default', () => {
      render(`<ais-refinement-list></ais-refinement-list>`);
      expect(createWidget.mock.calls[0][1].escapeFacetValues).toEqual(true);
    });
    afterEach(() => {
      createWidget.mockRestore();
    });
  });

  describe('[show more]/[show less] button', () => {
    const createFixture = (state, props) => {
      const fixture = render(state);
      Object.assign(fixture.componentInstance.testedWidget, props);
      fixture.detectChanges();
      return fixture;
    };
    it('should not show [show more] button when showMore === false', () => {
      const fixture = createFixture(
        {},
        {
          limit: 5,
        }
      );

      const showMoreBtn = fixture.debugElement.nativeElement.querySelector(
        'button'
      );
      expect(showMoreBtn).toBe(null);
    });

    it('should show [show more] button when showMore === true', () => {
      const fixture = createFixture(
        {
          toggleShowMore: jest.fn(),
        },
        {
          showMore: true,
          limit: 5,
          showMoreLabel: 'Please more',
          showLessLabel: 'Please less',
        }
      );

      const showMoreBtn = fixture.debugElement.nativeElement.querySelector(
        'button'
      );
      expect(showMoreBtn.innerHTML.trim()).toEqual('Please more');
    });

    it('should show [show more] button as disabled when canToggleShowMore === false', () => {
      const fixture = createFixture(
        {
          toggleShowMore: jest.fn(),
          canToggleShowMore: false,
        },
        {
          showMore: true,
          limit: 5,
          showMoreLabel: 'Please more',
          showLessLabel: 'Please less',
        }
      );

      const showMoreBtn = fixture.debugElement.nativeElement.querySelector(
        'button'
      );
      expect(showMoreBtn.disabled).toEqual(true);
    });

    it('should show [show more] button as enabled when canToggleShowMore === true', () => {
      const fixture = createFixture(
        {
          toggleShowMore: jest.fn(),
          canToggleShowMore: true,
        },
        {
          showMore: true,
          limit: 5,
          showMoreLabel: 'Please more',
          showLessLabel: 'Please less',
        }
      );

      const showMoreBtn = fixture.debugElement.nativeElement.querySelector(
        'button'
      );
      expect(showMoreBtn.disabled).toEqual(false);
    });

    it('should show [show less] button when isShowingMore === true', () => {
      const fixture = createFixture(
        {
          toggleShowMore: jest.fn(),
          isShowingMore: true,
        },
        {
          showMore: true,
          limit: 5,
          showMoreLabel: 'Please more',
          showLessLabel: 'Please less',
        }
      );

      const showMoreBtn = fixture.debugElement.nativeElement.querySelector(
        'button'
      );
      expect(showMoreBtn.innerHTML.trim()).toEqual('Please less');
    });

    it('should have ais-RefinementList-showMore CSS class', () => {
      const fixture = createFixture(
        {
          toggleShowMore: jest.fn(),
          canToggleShowMore: true,
        },
        {
          showMore: true,
          limit: 5,
          showMoreLimit: 10,
        }
      );

      const showMoreBtn = fixture.debugElement.nativeElement.querySelector(
        'button'
      );
      expect(
        showMoreBtn.classList.contains('ais-RefinementList-showMore')
      ).toBe(true);
    });

    it('should call `toggleShowMore()` on [show more] click', () => {
      const toggleShowMore = jest.fn();
      const fixture = createFixture(
        {
          toggleShowMore,
          canToggleShowMore: true,
        },
        {
          showMore: true,
          limit: 5,
          showMoreLimit: 10,
        }
      );
      const showMoreBtn = fixture.debugElement.nativeElement.querySelector(
        'button'
      );
      showMoreBtn.click();
      expect(toggleShowMore).toHaveBeenCalledTimes(1);
    });
  });
  it('should call `refine()` on item click', () => {
    const refine = jest.fn();
    const fixture = render({ refine });

    const [, secondItem] = fixture.debugElement.nativeElement.querySelectorAll(
      'li'
    );
    secondItem.click();

    expect(refine).toHaveBeenCalled();
    expect(refine).toHaveBeenCalledWith(defaultState.items[1].value);
  });

  it('should apply `transformItems` if specified', () => {
    const fixture = render({});
    fixture.componentInstance.testedWidget.transformItems = items =>
      items.map(item => ({ ...item, label: `transformed - ${item.label}` }));
    fixture.detectChanges();
    expect(fixture).toMatchSnapshot();
  });

  it('should call `searchForItems` when searching', () => {
    const searchForItems = jest.fn();
    const fixture = render({ searchForItems });

    // display search box
    fixture.componentInstance.testedWidget.searchable = true;
    fixture.detectChanges();

    const input = fixture.debugElement.nativeElement.querySelector('input');
    input.value = 'foobar';
    input.dispatchEvent(new Event('input'));

    expect(searchForItems).toHaveBeenCalled();
    expect(searchForItems).toHaveBeenCalledWith('foobar');
  });

  it('should be hidden with autoHideContainer', () => {
    const fixture = render({ items: [] });
    fixture.componentInstance.testedWidget.autoHideContainer = true;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
