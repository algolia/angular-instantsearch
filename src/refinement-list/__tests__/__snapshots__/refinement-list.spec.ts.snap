// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RefinementList renders markup with state 1`] = `
<ng-ais-refinement-list
  cx={[Function Function]}
  limitMin={[Function Number]}
  operator={[Function String]}
  platformId={[Function String]}
  searchInstance={[Function NgAisInstance]}
  searchPlaceholder={[Function String]}
  searchQuery=""
  showLessLabel={[Function String]}
  showMoreLabel={[Function String]}
  state={[Function Object]}
  updateState={[Function Function]}
>
  
      <div
    class="ais-RefinementList"
  >
    
        
    
    <div
      class="ais-RefinementList-body ais-body"
    >
      
          
      
      
  
          
      <ul
        class="ais-RefinementList-list"
      >
        
            
        
        <li
          class="ais-RefinementList-item"
        >
          
              
          <label
            class="ais-RefinementList-label"
          >
            
                
            <input
              class="ais-RefinementList-checkbox"
              type="checkbox"
              value="foo"
            />
            
                
            <ng-ais-highlight
              attributename="highlighted"
              ng-reflect-attribute-name="highlighted"
              ng-reflect-hit="[object Object]"
            >
              <span>
                foo
              </span>
            </ng-ais-highlight>
            
                
            <span
              class="ais-RefinementList-count"
            >
              
                  100
                
            </span>
            
              
          </label>
          
            
        </li>
        <li
          class="ais-RefinementList-item"
        >
          
              
          <label
            class="ais-RefinementList-label"
          >
            
                
            <input
              class="ais-RefinementList-checkbox"
              type="checkbox"
              value="bar"
            />
            
                
            <ng-ais-highlight
              attributename="highlighted"
              ng-reflect-attribute-name="highlighted"
              ng-reflect-hit="[object Object]"
            >
              <span>
                bar
              </span>
            </ng-ais-highlight>
            
                
            <span
              class="ais-RefinementList-count"
            >
              
                  100
                
            </span>
            
              
          </label>
          
            
        </li>
        <li
          class="ais-RefinementList-item"
        >
          
              
          <label
            class="ais-RefinementList-label"
          >
            
                
            <input
              class="ais-RefinementList-checkbox"
              type="checkbox"
              value="foobar"
            />
            
                
            <ng-ais-highlight
              attributename="highlighted"
              ng-reflect-attribute-name="highlighted"
              ng-reflect-hit="[object Object]"
            >
              <span>
                foobar
              </span>
            </ng-ais-highlight>
            
                
            <span
              class="ais-RefinementList-count"
            >
              
                  100
                
            </span>
            
              
          </label>
          
            
        </li>
        <li
          class="ais-RefinementList-item"
        >
          
              
          <label
            class="ais-RefinementList-label"
          >
            
                
            <input
              class="ais-RefinementList-checkbox"
              type="checkbox"
              value="barfoo"
            />
            
                
            <ng-ais-highlight
              attributename="highlighted"
              ng-reflect-attribute-name="highlighted"
              ng-reflect-hit="[object Object]"
            >
              <span>
                barfoo
              </span>
            </ng-ais-highlight>
            
                
            <span
              class="ais-RefinementList-count"
            >
              
                  100
                
            </span>
            
              
          </label>
          
            
        </li>
        
          
      </ul>
      
  
          
      
      
        
    </div>
    
      
  </div>
    
</ng-ais-refinement-list>
`;

exports[`RefinementList renders markup without state 1`] = `
<ng-ais-refinement-list
  cx={[Function Function]}
  limitMin={[Function Number]}
  operator={[Function String]}
  platformId={[Function String]}
  searchInstance={[Function NgAisInstance]}
  searchPlaceholder={[Function String]}
  searchQuery=""
  showLessLabel={[Function String]}
  showMoreLabel={[Function String]}
  state={[Function Object]}
  updateState={[Function Function]}
>
  
      <div
    class="ais-RefinementList"
  >
    
        
    
    <div
      class="ais-RefinementList-body ais-body"
    >
      
          
      
      
  
          
      <ul
        class="ais-RefinementList-list"
      >
        
            
        
        
          
      </ul>
      
  
          
      
      
        
    </div>
    
      
  </div>
    
</ng-ais-refinement-list>
`;

exports[`RefinementList should apply \`transformItems\` if specified 1`] = `
<ng-ais-refinement-list
  cx={[Function Function]}
  limitMin={[Function Number]}
  operator={[Function String]}
  platformId={[Function String]}
  searchInstance={[Function NgAisInstance]}
  searchPlaceholder={[Function String]}
  searchQuery=""
  showLessLabel={[Function String]}
  showMoreLabel={[Function String]}
  state={[Function Object]}
  transformItems={[Function Function]}
  updateState={[Function Function]}
>
  
      <div
    class="ais-RefinementList"
  >
    
        
    
    <div
      class="ais-RefinementList-body ais-body"
    >
      
          
      
      
  
          
      <ul
        class="ais-RefinementList-list"
      >
        
            
        
        <li
          class="ais-RefinementList-item"
        >
          
              
          <label
            class="ais-RefinementList-label"
          >
            
                
            <input
              class="ais-RefinementList-checkbox"
              type="checkbox"
              value="foo"
            />
            
                
            <ng-ais-highlight
              attributename="highlighted"
              ng-reflect-attribute-name="highlighted"
              ng-reflect-hit="[object Object]"
            >
              <span>
                transformed - foo
              </span>
            </ng-ais-highlight>
            
                
            <span
              class="ais-RefinementList-count"
            >
              
                  100
                
            </span>
            
              
          </label>
          
            
        </li>
        <li
          class="ais-RefinementList-itemais-RefinementList-item--selected"
        >
          
              
          <label
            class="ais-RefinementList-label"
          >
            
                
            <input
              class="ais-RefinementList-checkbox"
              type="checkbox"
              value="bar"
            />
            
                
            <ng-ais-highlight
              attributename="highlighted"
              ng-reflect-attribute-name="highlighted"
              ng-reflect-hit="[object Object]"
            >
              <span>
                transformed - bar
              </span>
            </ng-ais-highlight>
            
                
            <span
              class="ais-RefinementList-count"
            >
              
                  100
                
            </span>
            
              
          </label>
          
            
        </li>
        <li
          class="ais-RefinementList-item"
        >
          
              
          <label
            class="ais-RefinementList-label"
          >
            
                
            <input
              class="ais-RefinementList-checkbox"
              type="checkbox"
              value="foobar"
            />
            
                
            <ng-ais-highlight
              attributename="highlighted"
              ng-reflect-attribute-name="highlighted"
              ng-reflect-hit="[object Object]"
            >
              <span>
                transformed - foobar
              </span>
            </ng-ais-highlight>
            
                
            <span
              class="ais-RefinementList-count"
            >
              
                  100
                
            </span>
            
              
          </label>
          
            
        </li>
        <li
          class="ais-RefinementList-item"
        >
          
              
          <label
            class="ais-RefinementList-label"
          >
            
                
            <input
              class="ais-RefinementList-checkbox"
              type="checkbox"
              value="barfoo"
            />
            
                
            <ng-ais-highlight
              attributename="highlighted"
              ng-reflect-attribute-name="highlighted"
              ng-reflect-hit="[object Object]"
            >
              <span>
                transformed - barfoo
              </span>
            </ng-ais-highlight>
            
                
            <span
              class="ais-RefinementList-count"
            >
              
                  100
                
            </span>
            
              
          </label>
          
            
        </li>
        
          
      </ul>
      
  
          
      
      
        
    </div>
    
      
  </div>
    
</ng-ais-refinement-list>
`;

exports[`RefinementList should be hidden with autoHideContainer 1`] = `
<ng-ais-refinement-list
  autoHideContainer={[Function Boolean]}
  cx={[Function Function]}
  limitMin={[Function Number]}
  operator={[Function String]}
  platformId={[Function String]}
  searchInstance={[Function NgAisInstance]}
  searchPlaceholder={[Function String]}
  searchQuery=""
  showLessLabel={[Function String]}
  showMoreLabel={[Function String]}
  state={[Function Object]}
  updateState={[Function Function]}
>
  
      <div
    class="ais-RefinementList"
  >
    
        
    
    
      
  </div>
    
</ng-ais-refinement-list>
`;
